apps/web/pages/api/auth/[...nextauth].tsx

```diff
@@ -331,16 +331,16 @@ export default NextAuth({
     async signIn(params) {
       const { user, account, profile } = params;
 
-      if (account.provider === "email") {
+      if (account?.provider === "email") {
```

Comment:
Why not check for account instead of `?` every instance?

```suggestion
      if (!account) return false;

      if (account.provider === "email") {
```

Reply:
I don't know the expected values of account in all login scenario's (what is it when the provider is google, when credentials, etc) and also this would require me to at least check the flow of this entire function.

This approach was much easier development wise - undefined !== "oath" == return false, so that is certainly the right behaviour. This approach prevented having to test everything through as the logic remains self contained.

Reply:
Fair enough